pkg/blockstorage/azure/azuredisk.go

```diff
@@ -181,7 +184,14 @@ func (s *adStorage) SnapshotCopyWithArgs(ctx context.Context, from blockstorage.
 	blob := container.GetBlobReference(blobName)
 	defer deleteBlob(blob, blobName)
 
-	err = blob.Copy(*accessURI.AccessSAS, nil)
+	var copyOptions *storage.CopyOptions
+	if t, ok := ctx.Deadline(); ok {
+		time := uint(t.Second())
```

Comment:
Hmm... ctx.Deadline() returns a time.Time, an absolute time. t.Second() probably doesn't return what you expect. If you want absolute Unix time use t.Unix(), but I think you need a relative time (as per [Setting timeouts for Blob service operations](https://docs.microsoft.com/en-us/rest/api/storageservices/setting-timeouts-for-blob-service-operations)).

In this case I think you should be using  uint(t.Sub(time.Now()).Second()).

Reply:
This assumes that ctx.Deadline() is in the future....

Reply:
Looks like if the timeout is negative the Azure SDK won't care:

```
func addTimeout(params url.Values, timeout uint) url.Values {
	if timeout > 0 {
		params.Add("timeout", fmt.Sprintf("%v", timeout))
	}
	return params
}
```

Reply:
Actually timeout will not be negative since it is uint, I think they are just checking if it is not zero.

Reply:
Only after you cast it to uint will it lose the negativity :-)

Reply:
oops, my bad. updating the PR 🙈